test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs

```diff
@@ -101,10 +101,13 @@ public void TracerProviderSdkInvokesSamplingWithCorrectParameters()
             using (var fromInvalidW3CIdParent =
                 activitySource.StartActivity("customContext", ActivityKind.Client, "InvalidW3CIdParent"))
             {
-                // OpenTelemetry ActivityContext does not support
-                // non W3C Ids. Starting activity with non W3C Ids
-                // will result in no activity being created.
-                Assert.Null(fromInvalidW3CIdParent);
+                Assert.NotNull(fromInvalidW3CIdParent);
+                Assert.Equal("customContext", fromInvalidW3CIdParent.DisplayName);
+                Assert.Equal(ActivityKind.Client, fromInvalidW3CIdParent.Kind);
+
+                // OpenTelemetry ActivityContext does not support non W3C Ids.
+                // Starting activity with non W3C Ids will result in an Activity being created with no ParentId.
+                Assert.Null(fromInvalidW3CIdParent.ParentId);
```

Comment:
this is not doing what the test is claiming to be doing (Sampler receives correct params). If this seems irrelevant, consider removing this whole section.

Reply:
Sure, I was confused about what to Assert here.
The former code assumed the Activity would be null, so my asserts were focused on the Activity instance.

I debugged this again and verified that the Sampler was invoked and records the LatestSamplingParameters.
I'm updating this part.